src/backtrace/unwind.rs

```diff
@@ -20,60 +20,67 @@ static MISSING_DEBUG_INFO: &str = "debug information is missing. Likely fixes:
 
 /// Virtually* unwinds the target's program
 /// \* destructors are not run
-// FIXME(?) this should be "infallible" and return as many frames as possible even in case of IO
-// errors
+///
+/// This returns as much info as could be collected, even if the collection is interrupted by an error.
+/// if an error occurred during processing, it is stored in `Output::processing_error`.
 pub(crate) fn target(
     core: &mut Core,
     debug_frame: &[u8],
     vector_table: &VectorTable,
     sp_ram_region: &Option<RamRegion>,
-) -> anyhow::Result<Output> {
+) -> Output {
+    let mut output = Output {
+        corrupted: true,
+        outcome: Outcome::Ok,
+        raw_frames: vec![],
+        processing_error: None,
+    };
+
+    // returns all info collected until the error occurred and puts the error into `processing_error`
+    macro_rules! unwrap_or_return_output {
+        ( $e:expr ) => {
+            match $e {
+                Ok(x) => x,
+                Err(err) => {
+                    output.processing_error = Some(anyhow!(err));
+                    return output;
+                }
+            }
+        };
+    }
```

Comment:
I generally don't like function, struct or macro definitions inside of functions and would vote for moving this below the function.

Reply:
I generally agree, the reason I kept this in scope of the function is because it is awful enough as it is and I wanted to prevent anyone else from using it